mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -518,10 +518,20 @@ pub async fn run_taker_swap(swap: RunTakerSwapInput, ctx: MmArc) {
         }
         .fuse(),
     );
-    select! {
-        _swap = swap_fut => (), // swap finished normally
-        _touch = touch_loop => unreachable!("Touch loop can not stop!"),
-    };
+    // Run the swap in an abortable task and wait for it to finish.
+    let (swap_ended_notifier, swap_ended_notification) = oneshot::channel();
+    let abortable_swap = spawn_abortable(async move {
```

Comment:
Here we create an abortable future to run the swap loop (for using this the new stop rpc I believe).
But run_taker_swap fn, where we are now, is also wrapped in an abortable future. Could we use this one in the stop rpc?

Reply:
but how will we grab that abort handle and supply it to `run_taker_swap` in the first place?
also `kickstart_thread_handler` uses `run_taker_swap` inline (without spawning it), well eventually if u follow the usage ofc u will find `kickstart_thread_handler` is itself being spawned somewhere. but it's hard to propagate these handles all the way down to the swap runner methods.

Reply:
I think it's a bit odd when we use different futures to stop the same swap in different use cases (abort_all and stop_swap_rpc). Maybe we could get the handle for a future from the abortable_system for stop_swap_rpc, so we would use same abort technique in both cases. Maybe some modification for AbortableQueue is needed for that